tests/gtest/avifcodectest.cc

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "avif/avif.h"
+#include "aviftest_helpers.h"
+#include "gtest/gtest.h"
+
+namespace libavif {
+namespace {
+
+class CodecTest : public testing::TestWithParam<
+                      std::tuple</*encoding_codec=*/avifCodecChoice,
+                                 /*decoding_codec=*/avifCodecChoice>> {};
+
+TEST_P(CodecTest, EncodeDecode) {
+  const avifCodecChoice encoding_codec = std::get<0>(GetParam());
+  const avifCodecChoice decoding_codec = std::get<1>(GetParam());
+
+  if (avifCodecName(encoding_codec, AVIF_CODEC_FLAG_CAN_ENCODE) == nullptr ||
+      avifCodecName(decoding_codec, AVIF_CODEC_FLAG_CAN_DECODE) == nullptr) {
+    GTEST_SKIP() << "Codec unavailable, skip test.";
+  }
+
+  // AVIF_CODEC_CHOICE_SVT requires dimensions to be at least 64 pixels.
+  testutil::AvifImagePtr image =
+      testutil::CreateImage(/*width=*/64, /*height=*/64, /*depth=*/8,
+                            AVIF_PIXEL_FORMAT_YUV420, AVIF_PLANES_ALL);
+  ASSERT_NE(image, nullptr);
+  testutil::FillImageGradient(image.get());
+
+  testutil::AvifEncoderPtr encoder(avifEncoderCreate(), avifEncoderDestroy);
+  ASSERT_NE(encoder, nullptr);
+  encoder->codecChoice = encoding_codec;
```

Comment:
Unlike avifenc, the default encoder options of the C API tend to be "extreme" values (e.g., best quality/lossless) rather than the recommended values. So we may want to set some of the encoder options.

rav1e and SVT-AV1 don't support lossless compression (https://github.com/xiph/rav1e/issues/151 and https://gitlab.com/AOMediaCodec/SVT-AV1/-/issues/1636). So a quality of 100 is still not lossless if rav1e or SVT-AV1 is used.